### DISCUSSION TITLE
Mac OSX External Drive

### DISCUSSION THREAD

**Discussion by wts123123** [stashapp/stash#2958] [Q&A] [answered]
I have been trying to access an external drive in Mac OSX 12.5 (or 10.5 for that matter) but it seems to be unable to recognize the drive by the path I put in. 

In the configuration section, I try to add in the external volume by defining it in the 'Library' section. I am able to define the paths as `/.vol/DriveName` and `/Volumes/DriveName`. After this, I go to Tasks and then Library "Scan" but I don't get any responses in the "Task Queue" section. 

I grabbed the path from Disk Utility, it defines the mount path, but I have tried almost every version of the path I could think of including adding in root directory instead of folder specific. It doesn't pick up any files. If I attempt to scan my local folder there is no issue, it finds files.

I haven't seen anything specific in docs or discussions around Mac / osx specifically aside from install issues.

How do I properly define the external drive path for a mounted hard drive in osx?

**Comment by trytofly94** [comment]
I have the same Issue, OSX 13.0.1

**Reply by trytofly94** [reply]
Maybe someone has it working on MacOSX with an external drive? that way we would at least know that it is possible, and could proceed troubleshooting...

**Comment by KrisVardo** [comment]
I fought with this for a while, too.  I got it working by running the stash executable on the external drive also.  The executable gets loaded into RAM anyway and stays there because it's so small, so there's not a performance hit.

**Reply by stg-annon** [reply]
does MacOS do some kind of drive isolation that warrants doing this?

**Comment by KrisVardo** [comment]
Yes.  I imagine there's an Apple permissions-request library that Stash for OSX needs to leverage.  Stash needs to request Files and Folders access, or perhaps Full Disk Access.  I tried granting permission manually to Terminal and the shell but OSX did not recognize Stash as an executable that could be given permissions, nor does Stash request them when launched.

![Screen Shot 2022-12-11 at 10 21 40 AM](https://user-images.githubusercontent.com/114971575/206921539-b53a3d98-ad9e-42b4-b453-a525ba047657.png)

**Reply by stg-annon** [reply]
granting full disk access looks like it would be the solution here, what does it look like when you try to browse to the stash bin to grant it access?

**Reply by KrisVardo** [reply]
Well hey, it seems to support being granted access in V.18.  I hadn't tried it since v.16. 

![Screen Shot 2022-12-11 at 10 51 02 AM](https://user-images.githubusercontent.com/114971575/206922995-ba1fcf6a-370a-406c-85ba-e8e80e78c601.png)

@wts123123 you should try this and see if that makes the difference!

**Reply by ghost** [reply]
I granted full disk access with the ex on the external drive, but still nothing. It does the scan but does not return any results.

BEFORE I did that, I tested the scan after I put the external drive and got the follow errors;
```
WARN[2022-12-20 11:57:59] error scanning /Volumes/Drive: open /Volumes/Terrible: operation not permitted 
WARN[2022-12-20 11:57:59] error scanning /Volumes/Drive: open /Volumes/Terrible: operation not permitted 
```

After granting access, I did not get that error message. but it still doesn't work.

**Reply by ghost** [reply]
Okay, so it looks like one must grant terminal full disk access as well - doesn't prompt and error but I gave that a shot. Seems to be scanning now...

**Comment by KrisVardo** [comment]
And Terminal/shell access does not seem to grant Stash access by association.

**Comment by ghost** [comment, marked as answer]
Okay, so I think the fix is the following - it worked on my system;

1. Run the stashapp on the external drive
2. Grant full disk access for the stash app
3. Grant full disk access for terminal app

Worked for me. Thanks @KrisVardo and @stg-annon !
